### PR TITLE
[APM-820] new palette component

### PR DESCRIPTION
**Pull request by georgiana-roman**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://issues.alfresco.com/jira/browse/APM-820

**What is the new behaviour?**

component has been added 

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        return this.overlayPosition;
+    }
+
+    @HostListener('click', ['$event']) onClick($event) {
+        this.templatePortal = new TemplatePortal(this.templatePortalContent, this.amaPaleteContainerRef);
+        this.templatePortal.context = {$implicit: this.amaPaletteItem};
+        if (!this.amaPaletteOverlayRef.hasAttached()) {
+            this.amaPaletteOverlayRef.attach(this.templatePortal);
+        } else {
+            this.amaPaletteOverlayRef.detach();
```
Comment (popovicsandras) [inline src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts]: The overlay doesn't closes when you close anywhere else but the submenu buttons

File: src/app/process-editor/components/process-modeler/palette/palette.component.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if (!paletteItem.clickable) {
+            return;
+        }
+
+        if (paletteItem.group === 'tool') {
+            this.selectedTool = paletteItem;
+        }
+
+        this.delegateEvent(paletteItem, event);
+        this.paletteOverlayDirective.amaPaletteOverlayRef.detach();
```
Comment (popovicsandras) [inline src/app/process-editor/components/process-modeler/palette/palette.component.ts]: This just simply ruins the whole purpose of extracting the logic of overlaying into a directive. Please remove this 2 line and handle the detach from inside the directive:

Maybe it would be worth to consider this kind of hostListener also, or something similar:
```
    @HostListener('window:click', ['$event']) onWindowClick($event) {
        if (this.amaPaletteOverlayRef.hasAttached()) {
            this.amaPaletteOverlayRef.detach();
        }
    }
```

File: src/app/process-editor/components/process-modeler/palette/palette.component.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    selector: 'ama-process-palette'
+})
+export class PaletteComponent {
+
+    public selectedTool: ToolTrigger;
+    public paletteElements: PaletteElement[];
+    public opened = true;
+    public detach = false;
+    overlayRef: OverlayRef;
+    @ViewChild('drawer') templateContent: TemplateRef<any>;
```
Comment (popovicsandras) [inline src/app/process-editor/components/process-modeler/palette/palette.component.ts]: Hmm, can it handle multiple submenus also? :O
Because that one is a valid scenario also.

File: src/app/process-editor/config/palette-elements.json [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    },
+    {
+        "group": "element",
+        "type": "bpmn:CallActivity",
+        "icon": "bpmn-icon-call-activity",
+        "title": "APP.PROCESS_EDITOR.PALETTE.USER_TASK",
+        "clickable": true,
+        "draggable": true
+    },
+    {"group": "separator" },
```
Comment (popovicsandras) [inline src/app/process-editor/config/palette-elements.json]: From this line down, I assume we don't need those elements at the final product, they are just dummies.

File: src/app/process-editor/components/process-modeler/palette/palette.component.ts (hunk 3 above)
Comment (georgiana-roman) [reply src/app/process-editor/components/process-modeler/palette/palette.component.ts]: It should work, I tested this scenario.

File: src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ @Directive({
+     selector: '[ama-palette-overlay]'
+ })
+
+ export class PaletteOverlayDirective implements AfterViewInit {
+    overlayPosition: PositionStrategy;
+    templatePortal: TemplatePortal;
+    @Input() templatePortalContent: TemplateRef<any>;
+    @Input() amaPaletteItem: any;
+    @Input() amaPaletePortal: TemplatePortal;
```
Comment (popovicsandras) [inline src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts]: Do you use this Input at all? there is a typo, and because of it I'm not sure it is used.

File: src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    @Input() amaPaletteOverlayRef: OverlayRef;
+
+     constructor(
+        public submenuBtnRef: ElementRef,
+        public amaPaleteContainerRef: ViewContainerRef,
```
Comment (popovicsandras) [inline src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts]: typo: palette is with double `t`

File: src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+            {originX: 'start', originY: 'bottom'},
+            {overlayX: 'start', overlayY: 'bottom'}
+          );
+
+        return this.overlayPosition;
+    }
+
+    @HostListener('window:click', ['$event']) onWindowClick($event) {
+        if ($event.target === this.submenuBtnRef.nativeElement) {
+            this.templatePortal = new TemplatePortal(this.templatePortalContent, this.amaPaleteContainerRef);
```
Comment (popovicsandras) [inline src/app/process-editor/components/process-modeler/palette/palette-overlay.directive.ts]: typo here as well
